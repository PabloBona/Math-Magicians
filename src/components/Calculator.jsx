import React, { useState } from 'react';
import Display from './Display';
import RowButtons from './RowButtons';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calculatorData, setCalculatorData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  return (
    <div className="container">
      <div className="row my-3 text-light flex-column flex-md-row justify-content-center">
        <div className="col-md-6 my-3">
          <div className="d-flex align-items-center">
            {' '}
            <h2 className="align-self-md-center fs-4">
              Let&apos;s do some
              {' '}
              <span className="carrotColor">Maths!</span>
            </h2>
            <img src={`${process.env.PUBLIC_URL}/magician.png`} className="w-25" alt="Magician" />
          </div>
        </div>
        <div className="col-md-6 d-flex justify-content-center flex-wrap ">
          <Display calculatorData={calculatorData} />
          <RowButtons
            calculatorData={calculatorData}
            setCalculatorData={setCalculatorData}
            calculate={calculate}
          />
        </div>
      </div>
    </div>
  );
};

export default Calculator;
