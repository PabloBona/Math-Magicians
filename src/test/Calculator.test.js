import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Calculator component unit test', () => {
  test('Calculator Component Renders Properly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
