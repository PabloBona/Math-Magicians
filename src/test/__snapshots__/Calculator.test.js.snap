// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component unit test Calculator Component Renders Properly 1`] = `
<div
  className="container"
  data-testid="calculator"
>
  <div
    className="row my-3 text-light flex-column flex-md-row justify-content-center"
  >
    <div
      className="col-md-6 my-3"
    >
      <div
        className="d-flex align-items-center justify-content-center"
      >
         
        <h2
          className="align-self-md-center fs-4"
        >
          Let's do some
           
          <span
            className="carrotColor"
          >
            Maths!
          </span>
        </h2>
        <img
          alt="Magician"
          className="w-25"
          src="/magician.png"
        />
      </div>
    </div>
    <div
      className="col-md-6 d-flex justify-content-center flex-wrap "
    >
      <div
        className="col-8 col-6 d-flex justify-content-center"
      >
        <div
          className="d-flex justify-content-end text-light px-1 py-2 w-100 mx-5"
        >
          0
        </div>
      </div>
      <div
        className="col-8 col-6 d-flex justify-content-center flex-wrap w-50 bg"
      >
        <button
          className="btn btn-light rounded-0 border p-2 w-25 rounded-5 my-1 font-size "
          onClick={[Function]}
          type="submit"
        >
          AC
        </button>
        <button
          className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          ±
        </button>
        <button
          className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          %
        </button>
        <button
          className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          ÷
        </button>
        <button
          className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          7
        </button>
        <button
          className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          8
        </button>
        <button
          className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          9
        </button>
        <button
          className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          x
        </button>
        <button
          className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          4
        </button>
        <button
          className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          5
        </button>
        <button
          className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          6
        </button>
        <button
          className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          -
        </button>
        <button
          className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          1
        </button>
        <button
          className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          2
        </button>
        <button
          className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          3
        </button>
        <button
          className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          +
        </button>
        <button
          className="btn btn-light rounded-0 border py-2 w-50 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          0
        </button>
        <button
          className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          .
        </button>
        <button
          className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1"
          onClick={[Function]}
          type="submit"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
